supplemental_notebooks/2020-06-26_supplemental-notes.Rmd

```diff
@@ -0,0 +1,154 @@
+> For an explainer on FDR values, check out [this excellent video from StatQuest](https://www.youtube.com/watch?v=K8LQSvtjcEo).
+
+In practice, we're often interested in genes with low FDR values and the highest magnitude of log2 fold changes (LFC).
+(This is the rationale behind volcano plots like the one we made in the Intro to R and the Tidyverse module.)
+
+### Load in results from neuroblastoma cell line 
+
+Yesterday, we saved the object returned by `DESeq2::DESeq()` as an RDS file.
+
+```{r message=FALSE}
+deseq_object_file <- file.path("results", "NB_cell_line_DESeq_amplified_v_nonamplified.RDS")
+deseq_object <- readr::read_rds(deseq_object_file)
+```
+
+We need to load in the `DESeq2` library and we'll use the `%>%` operator later.
+
+```{r}
+library(DESeq2)
+`%>%` <- dplyr::`%>%`
+```
+
+### Shrunken log2 foldchanges
+
+There's an additional step we can take to get more accurate log2 fold change estimates to use for visualization or ranking.
+
+> We **highly recommend** taking a look at the resources we adapted this section from!
+>
+> * [Harvard Chan Bioinformatics Core (HBC). _DGE Analysis: Pairwise Comparisons (Wald Test)._](https://hbctraining.github.io/DGE_workshop_salmon/lessons/05_DGE_DESeq2_analysis2.html)
+> * [Love, Anders, and Huber. _Analyzing RNA-seq data with DESeq2._](https://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#log-fold-change-shrinkage-for-visualization-and-ranking)
+
+```{r}
+# Look at the contrasts available
+resultsNames(deseq_object)
+```
+
+We compared the _MYCN_ amplified cell lines to the non-amplified cell lines.
+Shrink the LFC for more accurate estimates.
+
+```{r}
+results_table_mycn <- results(deseq_object, 
+                              alpha = 0.05)
+results_lfc <- lfcShrink(deseq_object, 
+                         coef = "Status_Amplified_vs_Nonamplified",
+                         res = results_table_mycn)
+```
+
+```{r}
+results_lfc
+```
+
+Let's say we wanted to filter based on log2 fold change _and_ FDR before writing to file.
+You can use `dplyr::filter()` to filter based on the `log2FoldChange` and `padj` columns.
+Negative values in the `log2FoldChange` column will have lower expression in the Amplified group.
+If we did not take the absolute value of the `log2FoldChange` in our filtering step, we'd be missing out on those genes!
+
+```{r}
+results_lfc %>%
+  data.frame() %>%
+  tibble::rownames_to_column("Gene") %>%
+  dplyr::filter(padj < 0.05, abs(log2FoldChange) > 0.584)  # ~1.5x change
+```
+
+### Volcano plot
+
+We can make a volcano plot using a [Bioconductor](https://bioconductor.org/) package called [`EnhancedVolcano`](https://bioconductor.org/packages/release/bioc/html/EnhancedVolcano.html).
+Volcano plots provide information about the p-values (y-axis) and the log2 fold change (x-axis).
+
+```{r}
+# Here's how we install a Bioconductor package
+BiocManager::install("EnhancedVolcano")
```

Comment:
We should note that they should NOT update packages along with this. BioConductor will prompt them to update all the things.

Reply:
Also, testing this in a notebook, it will seem to hang while it waits for an answer, especially if the console is not visible, they may not know they need to enter anything there. I might say to run that in the console and type `n` OR `update=FALSE`